tests/phpunit/tests/privacy/wpPrivacyGeneratePersonalDataExportFile.php

```diff
@@ -213,6 +213,72 @@ public function test_detect_cannot_create_folder() {
 		wp_privacy_generate_personal_data_export_file( self::$export_request_id );
 	}
 
+	/**
+	 * @dataProvider data_export_data_grouped_invalid_type
+	 *
+	 * @ticket 51423
+	 *
+	 * @param mixed $groups '_export_data_grouped' post meta value.
+	 */
+	public function test_doing_it_wrong_for_export_data_grouped_invalid_type( $groups ) {
+		update_post_meta( self::$export_request_id, '_export_data_grouped', $groups );
+
+		$this->setExpectedIncorrectUsage( 'wp_privacy_generate_personal_data_export_file' );
+
+		wp_privacy_generate_personal_data_export_file( self::$export_request_id );
+	}
+
+	public function data_export_data_grouped_invalid_type() {
+		return array(
+			array( 10 ),
+			array( '10' ),
+			array( true ),
+			array( new stdClass() ),
```

Comment:
- Add `null` 
- Add `false`
- Change `'10'` to a `'foo'`
- Add serialized data string

Reply:
Resolved in commit https://github.com/WordPress/wordpress-develop/pull/758/commits/11361cad5a0e204c5e1ee7274ef0156d7e308d51